app/src/main/java/org/tomahawk/libtomahawk/utils/NetworkUtils.java

```diff
@@ -132,6 +134,21 @@ public static boolean isNetworkAvailable() {
         return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
     }
 
+    public static String getFinalURL(String url) throws IOException {
+        HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+        con.setInstanceFollowRedirects(false);
+        //Pretend we are a Browser, not Android something
+        con.setRequestProperty("User-Agent", "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; en-US; rv:1.9.2.2) Gecko/20100316 Firefox/3.6.2");
```

Comment:
Tbh, this part is completely unnecessary for this PR, here native player is used for amazon only and it is unnecessary to get final url for it. I can just remove this from here and see if I can come up with better approach in subsequent PRs, how does that sound?